document-style.md

```diff
@@ -0,0 +1,129 @@
+# Documentation style & reusable components
+
+The documentation for the project should maintain a consistent style where
+possible. This covers a number of aspects such as writing style, naming
+conventions, folder structure, links etc.
+
+All docs are currently in American English (en-US) but localization is planned.
+
+## Structure
+
+The main structure for documentation can be seen in `docs/docs`.
+There are a number of "root" sections:
+
+- `launch-manual` - For the current main Pulsar documentation
+- `packages` - Currently holds wiki info from the atom package repos
+- `resources` - For other referenced docs
+- `atom-archive` - For "as is" archived Atom documentation
+
+Within each section is an `index.md` which will usually contain info about each
+sub-section as well as links to them. These correspond to the second level 
+items on the sidebar.
+
+Inside `sections` are the sub-sections which group more specific topics. These
+also have an `index.md` which corresponds to the third level item on the
+sidebar. 
+This file is displayed on the website as a single long documenent but is
+actually created from a number of `@include()` lines which reference individual
```

Comment:
Only thing I'd add here is that we can do ./sections/section-name.md inside of this to prevent weird references using full directory paths. I know that should be obvious, but yea